osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, Dict, List, Union
+import uuid
+
+from osbenchmark.metrics import FileTestExecutionStore
+from osbenchmark import metrics, workload, config
+
+class Aggregator:
+    def __init__(self, cfg, test_executions_dict):
+        self.config = cfg
+        self.test_executions = test_executions_dict
+        self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
+        self.accumulated_iterations: Dict[str, int] = {}
+
+    # count iterations for each operation in the workload
+    def iterations(self) -> None:
```

Comment:
Nit: Might be better to rename this to `count_iterations` to be more descriptive

Reply:
An edge case: Line 20 is getting the default iterations from the workload but some users override this. For situations like this, we'll need to forgo the default iterations from the workload definition and count the iterations based on the count recorded in the results file:
```
 "test_procedure": "big5",
 "workload-params": {
  "iterations": 200, 
  "search_clients": 3000,
  "target_throughput": 3000
 }
```

Reply:
Had a sync with Michael offline: For now, will check if user overrode iterations. If not, will grab default from workload (as the code is doing here). Will address cases like `time-period` in future PR.

Reply:
Will also open issues to add configurable iterations to workloads, in order to handle workloads ran with custom iteration numbers as well.

Reply:
> Another conditional: After we verify that all test execution ids have the same test procedure, we'll need to add test_procedure to the config as well so that we can use that here to collect all the operations belonging to that test procedure.

If we already verified the test procedures are the same, shouldn't the operations be identical as well? I'm not sure what I would do with the collected operations

Reply:
> Nit: Might be better to rename this to be more descriptive. Something like `count_iterations` or `count_iterations_for_each_op` could work. If we go something descriptive like the later, we can also remove the comment above the method

Fixed!